### PR TITLE
chore(ci): add gpu tests from user documentation

### PR DESCRIPTION
**Pull request by soonum**

### CODE REVIEW

**Review by agnesLeroy — COMMENTED** [review]
Looks good to me

**Comment by agnesLeroy** [comment]
Hmm no, looks like the CPU tests try to execute the how_to_run_on_gpu doc test...

**Review by agnesLeroy — APPROVED** [review]

**Comment by soonum** [comment]
@slab-ci cpu_unsigned_integer_test

File: tfhe/src/test_user_docs.rs [hunk 1]
```diff
@@ -1,102 +1,116 @@
...
+fn test_gpu_doc() {
+    doctest!("../docs/how_to/run_on_gpu.md", how_to_run_on_gpu);
+}
+
+fn main() {
+    #[cfg(not(feature = "gpu"))]
+    test_cpu_doc();
+    #[cfg(feature = "gpu")]
+    test_gpu_doc()
+}
```
Comment (IceTDrinker) [inline tfhe/src/test_user_docs.rs:116]: main ?

**Review by IceTDrinker — COMMENTED** [review]
I don't think we should merge with a main in the test_user_docs.rs file ? 🤔

File: tfhe/src/test_user_docs.rs (hunk 1 above)
Comment (IceTDrinker) [reply tfhe/src/test_user_docs.rs:116]: I think we are not running any doctests here ? as main is a private function called nowhere ?

**Review by IceTDrinker — CHANGES_REQUESTED** [review]
Requesting change as I'm really unsure on that main/doc change

File: tfhe/src/test_user_docs.rs [hunk 2]
```diff
@@ -1,102 +1,109 @@
...
-// Tutorials
-doctest!(
-    "../docs/tutorials/ascii_fhe_string.md",
-    ascii_fhe_string_tutorial
-);
-doctest!("../docs/tutorials/parity_bit.md", parity_bit_tutorial);
+#[cfg(feature = "gpu")]
+fn test_gpu_doc() {
+    doctest!("../docs/how_to/run_on_gpu.md", how_to_run_on_gpu);
+}
```
Comment (IceTDrinker) [inline tfhe/src/test_user_docs.rs:109]: I think it should be a module

like

```rust
#[cfg(featyre = "gpu")]
mod test_gpu_user_doc {
    doctest!(...);
}
```

the same for cpu but with the

#[cfg(not(feature = "gpu"))]

as the functions are not called, the doctest! macro creates the entry points

Comment (soonum) [reply tfhe/src/test_user_docs.rs]: You're right

**Review by IceTDrinker — APPROVED** [review]

**Pull request merged**